MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,77 @@
+# Migration Guide
+
+## Upgrading from v0.46.0 → v0.47.0
+
+There are deprecations in this update. Please ensure you read this guide thoroughly and prepare your potential 
+automated workflows before upgrading.
+
+### Deprecations
+
+- [User Roles](#user-roles)
+
+### User Roles
+
+The `roles` field on the `auth0_user` resource will continue to be available for managing user roles. However, to ensure
```

Comment:
Adding a migration guide as we are deprecating the roles field on the user resource.